index.d.ts

```diff
@@ -0,0 +1,176 @@
+declare module "react-native-square-reader-sdk" {
```

Comment:
@caleb-harrelson appreciate the fix here, but we'd like to avoid reverting the existing TypeScript implementation unless there is a strong reason for that given't we don't want to maintain Javascript code any more. 

There are a few fixes done to the existing typescript solution. Please let me know if you feel strongly about this change. Otherwise, I'd suggest to close this PR.

Reply:
@hukid, what do you mean by not wanting to maintain JavaScript code anymore? That you intend to switch the library to TypeScript or that the library is no longer being supported?

I just want to be clear that there is no "existing TypeScript implementation" in this library. The previous PR claimed to add it, but did not do so. This PR doesn't implement TypeScript either, but would expose the library's functions (still written in JavaScript) to consumers using TypeScript.

Reply:
@caleb-harrelson I believe I see your point and I agree with you. Now I realize the original change you are reverting only create typescript for unit test but failed to expose the TypeScript interface to actual developers. Sorry for the misunderstanding and appreciate the patiently explanation. 

Your point makes sense and I'll approve this PR assuming you'll resolve the conflict. cc: @CharmisG.

Reply:
Great, thanks!

I've resolved the conflict and rebased.

Reply:
Thanks a lot! Do you have permission to merge?